src/3_branch.js

```
@@ -1123,27 +1181,52 @@ if (config.WEB_BUILD) {
-			description: options['description'] || '',
-			openAppButtonText: options['openAppButtonText'] || 'View in app',
-			downloadAppButtonText: options['downloadAppButtonText'] || 'Download App',
-			iframe: typeof options['iframe'] == 'undefined' ? true : options['iframe'],
-			showiOS: typeof options['showiOS'] == 'undefined' ? true : options['showiOS'],
-			showAndroid: typeof options['showAndroid'] == 'undefined' ? true : options['showAndroid'],
-			showDesktop: typeof options['showDesktop'] == 'undefined' ? true : options['showDesktop'],
-			disableHide: !!options['disableHide'],
-			forgetHide: !!options['forgetHide'],
-			makeNewLink: !!options['make_new_link']
-		};
+		var self = this;
+		this._queue(function(next) {
+			var bannerOptions = {
+				icon: options['icon'] || '',
+				title: options['title'] || '',
+				description: options['description'] || '',
+				openAppButtonText: options['openAppButtonText'] || 'View in app',
+				downloadAppButtonText: options['downloadAppButtonText'] || 'Download App',
+				iframe: typeof options['iframe'] == 'undefined' ? true : options['iframe'],
+				showiOS: typeof options['showiOS'] == 'undefined' ? true : options['showiOS'],
+				showAndroid: typeof options['showAndroid'] == 'undefined' ? true : options['showAndroid'],
+				showDesktop: typeof options['showDesktop'] == 'undefined' ? true : options['showDesktop'],
+				disableHide: !!options['disableHide'],
+				forgetHide: !!options['forgetHide'],
+				make_new_link: !!options['make_new_link']
+			};
 
-		if (typeof options['showMobile'] != 'undefined') {
-			bannerOptions.showiOS = bannerOptions.showAndroid = options['showMobile'];
-		}
+			if (typeof options['showMobile'] != 'undefined') {
+				bannerOptions.showiOS = bannerOptions.showAndroid = options['showMobile'];
+			}
 
-		banner(this, bannerOptions, linkData, this._storage);
+			banner(self, bannerOptions, linkData, self._storage);
+			next();
+		});
+	};
+
+	Branch.prototype['closeBanner'] = function() {
```

hrm, i think there may be an argument for having this always defined, and just fail silently when the banner is open... but i could be wrong